pkg/reconciler/channel/channel.go

```diff
@@ -98,28 +110,51 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 	return reconcileErr
 }
 
-func (r *Reconciler) reconcile(ctx context.Context, ch *v1alpha1.Channel) error {
-	// Do not Initialize() Status in channel-default-controller. It will set ChannelConditionProvisionerInstalled=True
-	// Directly call GetCondition(). If the Status was never initialized then GetCondition() will return nil and
-	// IsUnknown() will return true
-	c := ch.Status.GetCondition(v1alpha1.ChannelConditionProvisionerInstalled)
+func (r *Reconciler) reconcile(ctx context.Context, c *v1alpha1.Channel) error {
+	c.Status.InitializeConditions()
 
-	if c == nil || c.IsUnknown() {
+	// 1. Create the backing Channel CRD, if it doesn't exist.
+	// 2. Patch the subscriptions from this Channel into the backing Channel CRD.
+	// 3. Propagate the backing Channel CRD Status, Address, and SubscribableStatus into this Channel.
 
-		var proName string
-		var proKind string
-		if ch.Spec.Provisioner != nil {
-			proName = ch.Spec.Provisioner.Name
-			proKind = ch.Spec.Provisioner.Kind
-		}
+	if c.DeletionTimestamp != nil {
+		// Everything is cleaned up by the garbage collector.
+		return nil
+	}
+
+	channelResourceInterface := r.DynamicClientSet.Resource(duckroot.KindToResource(c.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())).Namespace(c.Namespace)
+
+	// Tell tracker to reconcile this Channel whenever the backing Channel CRD changes.
+	track := r.resourceTracker.TrackInNamespace(c)
+
+	backingChannel, err := r.reconcileBackingChannel(ctx, channelResourceInterface, c)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem reconciling the backing channel", zap.Error(err))
```

Comment:
Looking at the logs, we end up seeing these error messages repeated many times. How about, rather than logging at each level, we prepend something to the error and return it? The `Reconcile` will log error.

```golang
return fmt.Errorf("problem reconciling the backing channel: %v", err)
```

Thoughts?

Reply:
Makes sense, done, thanks!